File: app/src/main/java/com/aar/app/wordsearch/features/gameover/GameOverViewModel.java
package com.aar.app.wordsearch.features.gameover;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.arch.lifecycle.ViewModel;

import com.aar.app.wordsearch.data.GameDataSource;
import com.aar.app.wordsearch.model.GameDataInfo;

public class GameOverViewModel extends ViewModel {

    private GameDataSource mGameDataSource;
    private MutableLiveData<GameDataInfo> mOnGameDataInfoLoaded = new MutableLiveData<>();

    public GameOverViewModel(GameDataSource gameDataSource) {
        mGameDataSource = gameDataSource;
    }

    public void loadData(int gid) {
        mGameDataSource.getGameDataInfo(gid, mOnGameDataInfoLoaded::setValue);
    }

    public void deleteGameRound(int gid) {
        mGameDataSource.deleteGameData(gid);
    }

    public LiveData<GameDataInfo> getOnGameDataInfoLoaded() {
        return mOnGameDataInfoLoaded;
    }
}

File: app/src/main/java/co/africanwolf/hiddenwords/gameover/GameOverViewModel.kt
package co.africanwolf.hiddenwords.gameover

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import co.africanwolf.hiddenwords.data.sqlite.GameDataSQLiteDataSource
import co.africanwolf.hiddenwords.model.GameDataInfo
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class GameOverViewModel @Inject constructor(private val mGameDataSource: GameDataSQLiteDataSource) : ViewModel() {
    private val mOnGameDataInfoLoaded = MutableLiveData<GameDataInfo>()
    fun loadData(gid: Int) {
        mGameDataSource.getGameDataInfo(gid) { value: GameDataInfo -> mOnGameDataInfoLoaded.setValue(value) }
    }

    fun deleteGameRound(gid: Int) {
        mGameDataSource.deleteGameData(gid)
    }

    val onGameDataInfoLoaded: LiveData<GameDataInfo>
        get() = mOnGameDataInfoLoaded
}
